gems/aws-crt/native/src/signing_config.c

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include "crt.h"
+
+#include <aws/auth/credentials.h>
+#include <aws/auth/signing_config.h>
+#include <aws/common/string.h>
+
+struct aws_crt_signing_config {
+    struct aws_signing_config_aws native;
+    struct aws_string *region_str;
+    struct aws_string *service_str;
+    struct aws_string *signed_body_value_str;
+};
+
+struct aws_crt_signing_config *aws_crt_signing_config_new(
+    int algorithm,
+    int signature_type,
+    const char *region,
+    const char *service,
+    const char *signed_body_value,
+    uint64_t date_epoch_ms,
+    struct aws_credentials *credentials,
+    int aws_signed_body_header_type,
+    aws_should_sign_header_fn *should_sign_header,
+    bool use_double_uri_encode,
+    bool should_normalize_uri_path,
+    bool omit_session_token,
+    uint64_t expiration_in_seconds) {
+    struct aws_allocator *allocator = aws_crt_allocator();
+    struct aws_crt_signing_config *config = aws_mem_acquire(allocator, sizeof(struct aws_crt_signing_config));
+
+    AWS_ZERO_STRUCT(*config);
```

Comment:
trivial: aws_mem_calloc() instead of aws_mem_acquire() followed by AWS_ZERO_STRUCT